edge-modules/MetricsCollector/MetricWithCustomTags.cs

```diff
@@ -0,0 +1,18 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Agent.Diagnostics
+{
+    using System;
+    using System.Collections.Generic;
+    using Microsoft.Azure.Devices.Edge.Util;
+
+    public sealed class MetricWithCustomTags : Metric, IEquatable<Metric>
```

Comment:
Metric class already implement IEquatable<Metric>, why do you need to explicitly implement it here?
```suggestion
    public sealed class MetricWithCustomTags : Metric
```